[Feature] Styleguidist overhaul (#766)
vrk-kpa/suomifi-ui-components: src/core/Expander/Expander/Expander.md

  </ExpanderTitleButton>
  <ExpanderContent>
    <Paragraph marginBottomSpacing="l">
      If your company is experiencing financial or payment
      difficulties, please contact the Yrittäjän talousapu counselling
      service.
    </Paragraph>
    <Paragraph marginBottomSpacing="l">
      Telephone Service 0295 024 882
    </Paragraph>
    <Paragraph>Monday - Friday 9.00–16.15</Paragraph>
  </ExpanderContent>
</Expander>;
```

### Default open

Use the `defaultOpen` prop to make the Expander show its content on first render.

```jsx
import {
  Expander,
  ExpanderTitleButton,
  ExpanderContent,
  Paragraph
} from 'suomifi-ui-components';

<Expander defaultOpen>
  <ExpanderTitleButton>
    Assistance with financial problems
  </ExpanderTitleButton>
  <ExpanderContent>
    <Paragraph marginBottomSpacing="l">
      If your company is experiencing financial or payment
      difficulties, please contact the Yrittäjän talousapu counselling
      service.
    </Paragraph>
    <Paragraph marginBottomSpacing="l">
      Telephone Service 0295 024 882
    </Paragraph>
    <Paragraph>Monday - Friday 9.00–16.15</Paragraph>
  </ExpanderContent>
</Expander>;
```

### Heading semantics

> You can use the `asHeading` prop in ExpanderTitleButton to render the Expander button as a heading element.

Reviewer: To clarify, this was prompted by an a11y review where I saw ExpanderGroup being used with lots of content with subheadings. When multiple sections are expanded simultaneously, all those subheadings will then appear to be on the same level with each other.

  </Expander>
  <Expander>
    <ExpanderTitleButton>
      Errors in the Suomi.fi mobile application
    </ExpanderTitleButton>
    <ExpanderContent>
      <Paragraph marginBottomSpacing="l">
        If the application does not work on your mobile device for
        some reason, you can also read your messages via a browser
      </Paragraph>
      <Paragraph>
        Errors occurring in the Suomi.fi application generally do not
        affect the preservation of Suomi.fi Messages.
      </Paragraph>
    </ExpanderContent>
  </Expander>
</ExpanderGroup>;
```

### Controlled state

You can control the state of an individual Expander programmatically with the `open` prop.

```jsx
import {
  Expander,
  ExpanderGroup,
  ExpanderTitleButton,
  ExpanderContent,
  Paragraph,
  Text
} from 'suomifi-ui-components';
import { useState } from 'react';

const [expanderOneOpen, setExpanderOneOpen] = useState(false);
const [expanderTwoOpen, setExpanderTwoOpen] = useState(false);
const [expanderThreeOpen, setExpanderThreeOpen] = useState(true);

<>
  <button onClick={() => setExpanderTwoOpen(!expanderTwoOpen)}>
    Toggle middle expader
  </button>
  <ExpanderGroup
    openAllText="Open all"

>     ariaOpenAllText="Open all expanders"
>     closeAllText="Close all"
>     ariaCloseAllText="Close all expanders"

Reviewer: Because I doubt "expanders" is ever a word any user would understand, I suggest using "sections" as a default suggestion.
```suggestion
    ariaOpenAllText="Open all sections"
    closeAllText="Close all"
    ariaCloseAllText="Close all sections"
```

Reviewer 2: An excellent point 👍

              who can access your data.
            </li>
          </ul>
        </Paragraph>
        <Paragraph>
          If you break your mobile device or lose it, you can sign out
          of the Suomi.fi application either in the Suomi.fi web
          service or by using another phone.
        </Paragraph>
      </ExpanderContent>
    </Expander>
    <Expander open={expanderThreeOpen}>
      <ExpanderTitleButton>
        Errors in the Suomi.fi mobile application
      </ExpanderTitleButton>
      <ExpanderContent>
        <Paragraph marginBottomSpacing="l">
          If the application does not work on your mobile device for
          some reason, you can also read your messages via a browser
        </Paragraph>
        <Paragraph>
          Errors occurring in the Suomi.fi application generally do
          not affect the preservation of Suomi.fi Messages.
        </Paragraph>
      </ExpanderContent>
    </Expander>
  </ExpanderGroup>
</>;
```

### Interactive elements in Expander title

If you need to have interactive elements in the Expander title, use `<ExpanderTitle>` instead of `<ExpanderTitleButton>`.

> ExpanderTitle creates a smaller toggle button to the right side of the Expander. It is important to set `ariaOpenText`/`ariaCloseText` props as well as connect the interactive element to the button with `toggleButtonAriaDescribedBy` to provide adequate context for assistive technologies.

Reviewer: Because aria-expanded already communicates state, it can be confusing to change the text too. This has been reported separately as a bug of the component itself, but may I suggest a temporary "fix" of simply making the example text the same?

 ```suggestion
ExpanderTitle creates a smaller toggle button to the right side of the Expander. Generally `ariaOpenText`/`ariaCloseText` props should have the same value, because state is already communicated to assistive technology using `aria-expanded`. Point `toggleButtonAriaDescribedBy` to the ID of interactive element to provide additional context for assistive technology.
```

Author: I say let's go with this docs change for now. 

Changed it like you suggested.

```jsx
import {
  Expander,
  ExpanderTitle,
  ExpanderContent,
  Checkbox,
  Paragraph
} from 'suomifi-ui-components';

<Expander>
  <ExpanderTitle

>     ariaOpenText="Show additional information"
>     ariaCloseText="Close additional information"

Reviewer: ```suggestion
    ariaOpenText="Additional information"
    ariaCloseText="Additional information"
```

Author: Done
